verifier/src/election/verifiers/chained-hash.verifier.js

```diff
@@ -0,0 +1,53 @@
+const chalk = require("chalk");
+
+const { hash } = require("../../crypto-utils");
+const { readAllElectionFiles } = require("../../file-utils");
+
+/**
+ * A verifier that checks if all the `chainedHash` property in the log entries
+ * are computed properly.
+ *
+ * @module
+ */
+module.exports = {
+  /**
+   * Whether the chained hashes are correct or not.
+   *
+   * @param {String} path - a directory that contains all the extracted files in a election log file.
+   * @returns {Promise<Boolean>}
+   */
+  verifyChainedHashes(path) {
+    let previousHash = null;
+
+    return new Promise((resolve) => {
+      readAllElectionFiles(
+        path,
+        (data, fail) => {
+          const { signed_data: signedData, chained_hash: chainedHash } = data;
+          // The first hash is computed using the election unique id.
+          if (!previousHash) {
+            const { message_id: messageId } = data;
+            const [authorityName, electionId] = messageId.split(".");
+            previousHash = [authorityName, electionId].join(".");
```

Comment:
I must be not seeing something somewhere, but this to me looks like
```
previousHash = messageId
```

Reply:
@orlera The `messageId` is longer 😅  so I am just taking the first two elements!